src/platform/ESP32/bluedroid/BLEManagerImpl.cpp

```diff
@@ -1009,6 +1048,73 @@ void BLEManagerImpl::HandleTXCharCCCDRead(esp_ble_gatts_cb_param_t * param)
     }
 }
 
+#ifdef CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+void BLEManagerImpl::HandleC3CharRead(esp_ble_gatts_cb_param_t * param)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    CHIPoBLEConState * conState;
+    esp_gatt_rsp_t rsp;
+
+    chip::System::PacketBufferHandle bufferHandle;
+    BitFlags<AdditionalDataFields> additionalDataFields;
+    AdditionalDataPayloadGeneratorParams additionalDataPayloadParams;
+
+    // Data is sent in the chunk of size (ESP_GATT_DEF_BLE_MTU_SIZE - 1)
+    uint8_t chunkLen = ESP_GATT_DEF_BLE_MTU_SIZE - 1;
```

Comment:
Should this be used or the negotiated MTU (from the MTU exchange message)?

Reply:
Thought the same, but did not receive an event for MTU negotiation, thats why using default.

Reply:
Is `ESP_GATTS_MTU_EVT` checked/added here? It should have the MTU, if negotiated, in `param->mtu.mtu`

Reply:
It is there, I changed the log level to check if we are hitting this code path but code flow never came in here.

https://github.com/project-chip/connectedhomeip/blob/d08e56e679fb8e2430169ae01ae94b2ed4db1e30/src/platform/ESP32/bluedroid/BLEManagerImpl.cpp#L900-L904